app/routes/submissions.rb

```diff
@@ -15,8 +15,15 @@ class Submissions < Core
 
         title("%s by %s in %s" % [submission.problem.name, submission.user.username, submission.problem.language])
 
+        comment_history = if current_user.guest? || !$flipper[:comment_history].enabled?(current_user)
```

Comment:
The main benefit was being able to test that the data returned for a feature was correct/expected, same when the feature was not enabled. If I remember correctly I also made something like `current_user.feature?(:my_feature)` which returned the appropriate permission for an user or a guest. Not really saying this would work in exercism context though, just sharing experiences. 😄 

The presentation layer is always complicated, I relied on presenters to do the trick when the feature changed the presentation layer too much, but in this case I think it would be overkill. 🤔